### PR TITLE
Update Ubuntu images in `tools/sdk-generation-pipeline` to Ubuntu 22.04

### PR DESCRIPTION
**Pull request by konrad-jamrozik**
This is a companion PR to:
- #4996

In this PR, I migrate all occurrences of Ubuntu to `22.04` in the directory `tools/sdk-generation-pipeline`.

This PR is part of work on:
- #5178

This PR requires review by @dw511214992, notably because of this comment:
- https://github.com/Azure/azure-sdk-tools/pull/4996#discussion_r1052977794

@dw511214992 could you let me know what else needs to be done so that we can migrate `tools/sdk-generation-pipeline` to Ubuntu 22.04? For example, how to migrate the docker image? Could I assist somehow with this? Note we want to do this migration to avoid issues like #4888 and get all the usual benefits on staying up to date with the newest image.

Also please note this PR currently causes [the build to fail on Go integration tests](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2069511&view=logs&j=46f7c609-157b-5ecf-ad71-5099ca1c8979&t=d8ca8b56-f5e3-544d-457e-eba43fd35a30). I would appreciate any assistance with fixing this failure.

### CODE REVIEW

File: tools/sdk-generation-pipeline/Dockerfile [hunk 1]
```diff
@@ -31,7 +31,7 @@ RUN apt install python3-pip -y && apt install python3-venv -y && pip3 install --
 RUN add-apt-repository ppa:deadsnakes/ppa -y && apt-get install python3.10 -y && apt-get install python3.10-venv -y
 
 # install powershell
-RUN wget -q https://packages.microsoft.com/config/ubuntu/20.04/packages-microsoft-prod.deb
+RUN wget -q https://packages.microsoft.com/config/ubuntu/22.04/packages-microsoft-prod.deb
```
Comment (konrad-jamrozik) [inline tools/sdk-generation-pipeline/Dockerfile:34]: There is additional context for this change here:
- https://github.com/Azure/azure-sdk-tools/pull/4996#discussion_r1052977794

File: tools/sdk-generation-pipeline/.github/workflows/ci.yml [hunk 2]
```diff
@@ -6,7 +6,7 @@ on:
     branches: [ "main" ]
 jobs:
   build:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-22.04
```
Comment (konrad-jamrozik) [inline tools/sdk-generation-pipeline/.github/workflows/ci.yml:9]: Note this is not a 1ES-hosted image, instead it is a [GitHub image](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#choosing-github-hosted-runners).

File: tools/sdk-generation-pipeline/Dockerfile (hunk 1 above)
Comment (dw511214992) [reply tools/sdk-generation-pipeline/Dockerfile:34]: Please revert this change because it's in our docker image, which is based on ubuntu 20.04.

File: tools/sdk-generation-pipeline/documents/task-engine/README.md [hunk 3]
```diff
@@ -20,7 +20,7 @@ SDK Automation is launched in azure pipeline. It runs tasks in the following ste
 
 ### CodegenToSdkConfig
 This is type of file `./codegen_to_sdk_config.json` in sdk repo.
-The running environment of these scripts would be expected to be __Ubuntu 20.04__ on Azure Pipeline. This may change in the future. All the running script should be executable.
+The running environment of these scripts would be expected to be __Ubuntu 22.04__ on Azure Pipeline. This may change in the future. All the running script should be executable.
```
Comment (dw511214992) [inline tools/sdk-generation-pipeline/documents/task-engine/README.md:23]: Please revert this change because our docker image is based on ubuntu 20.04.

File: tools/sdk-generation-pipeline/Dockerfile (hunk 1 above)
Comment (konrad-jamrozik) [reply tools/sdk-generation-pipeline/Dockerfile:34]: @dw511214992 is there a way for us to update it to ubuntu 22.04? What steps it would be necessary for that?

Thanks!

File: tools/sdk-generation-pipeline/documents/task-engine/README.md (hunk 3 above)
Comment (konrad-jamrozik) [reply tools/sdk-generation-pipeline/documents/task-engine/README.md:23]: @dw511214992 please check out [my comment here](https://github.com/Azure/azure-sdk-tools/pull/5004#discussion_r1055082694). Thanks!

File: tools/sdk-generation-pipeline/Dockerfile (hunk 1 above)
Comment (weshaggard) [reply tools/sdk-generation-pipeline/Dockerfile:34]: @dw511214992 we are trying to move most of our agents to run on 22.04 so we should please take a tracking issue to update these to 22.04.

File: tools/sdk-generation-pipeline/.github/workflows/ci.yml (hunk 2 above)
Comment (weshaggard) [reply tools/sdk-generation-pipeline/.github/workflows/ci.yml:9]: @dw511214992 Do we use this github action anywhere? If so we should try to remain pinned to a version instead of floating via latest ad things tend to break when the move that reference.

@konrad-jamrozik this is github actions which currently doesn't support our 1ES agents.

Comment (konrad-jamrozik) [reply tools/sdk-generation-pipeline/.github/workflows/ci.yml:9]: @weshaggard thanks, that's what I thought. Just to clarify: I believe the change should work, as it uses the [GitHub-hosted Ubuntu 22.04](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#choosing-github-hosted-runners), not 1ES agent.

Comment (weshaggard) [reply tools/sdk-generation-pipeline/.github/workflows/ci.yml:9]: Yes the image update should be fine but we just cannot use our 1ES pool.

File: tools/sdk-generation-pipeline/Dockerfile (hunk 1 above)
Comment (konrad-jamrozik) [reply tools/sdk-generation-pipeline/Dockerfile:34]: @dw511214992 kindly asking for input on this :)

File: tools/sdk-generation-pipeline/.github/workflows/ci.yml (hunk 2 above)
Comment (konrad-jamrozik) [reply tools/sdk-generation-pipeline/.github/workflows/ci.yml:9]: @dw511214992 kindly asking for input on this :)

**Comment by konrad-jamrozik** [comment]
@dw511214992 @weshaggard I am closing this PR as "blocked". I created an issue for @dw511214992 to follow up when appropriate:
- #5178

Thanks!

File: tools/sdk-generation-pipeline/documents/task-engine/README.md (hunk 3 above)
Comment (konrad-jamrozik) [reply tools/sdk-generation-pipeline/documents/task-engine/README.md:23]: I created a follow-up issue. Please see this comment:
- https://github.com/Azure/azure-sdk-tools/pull/5004#issuecomment-1398536891

File: tools/sdk-generation-pipeline/.github/workflows/ci.yml (hunk 2 above)
Comment (konrad-jamrozik) [reply tools/sdk-generation-pipeline/.github/workflows/ci.yml:9]: I created a follow-up issue. Please see this comment:
- https://github.com/Azure/azure-sdk-tools/pull/5004#issuecomment-1398536891

File: tools/sdk-generation-pipeline/Dockerfile (hunk 1 above)
Comment (konrad-jamrozik) [reply tools/sdk-generation-pipeline/Dockerfile:34]: I created a follow-up issue. Please see this comment:
- https://github.com/Azure/azure-sdk-tools/pull/5004#issuecomment-1398536891
